spec/toggle-quotes-spec.js

```diff
@@ -184,6 +184,31 @@ describe('ToggleQuotes', () => {
         })
       })
     })
+
+    describe('when the string contains escaped characters', () => {
+      it('does not remove them if they are at the end of the line', () => {
+        editor.setText(`'\\\\'`)
+        editor.setCursorBufferPosition([0, 2])
+        toggleQuotes(editor)
+        expect(editor.getText()).toBe(`"\\\\"`)
+      })
+
+      describe('when there is an escaped quote mark', () => {
+        it('unescapes if it is the original delim', () => {
+          editor.setText(`'\\\''`)
+          editor.setCursorBufferPosition([0, 3])
+          toggleQuotes(editor)
+          expect(editor.getText()).toBe(`"'"`)
```

Comment:
When I try this out in Atom, I get a different result:

![demo](https://user-images.githubusercontent.com/2988/56670694-4f5fbf00-6681-11e9-940e-4be0ee0c059d.gif)

The behavior I'm seeing in Atom seems like the right behavior to me, but I'm confused as to why the test seems to be asserting something different. Can you help me understand why the behavior in Atom appears to contradict the behavior asserted in the test? (Perhaps I'm misreading the test in some way?)

Reply:
In the tests, the backslashes need to be escaped. So I believe what you should actually be testing in Atom is `'\''`.

(The test also seems to have an unnecessary escape before the single quote)

Reply:
I've removed the redundant `\` in the source code.